build.gradle.kts

```diff
@@ -31,23 +32,20 @@ if (matchResult != null) {
 }
 val releaseVersion = project.findProperty("releaseVersion") as String? ?: snapshotVersion
 
-val bufCLIFile = project.layout.buildDirectory.file("gobin/buf").get().asFile
-val bufCLIPath: String = bufCLIFile.absolutePath
+val buf: Configuration by configurations.creating
 val bufLicenseHeaderCLIFile = project.layout.buildDirectory.file("gobin/license-header").get().asFile
 val bufLicenseHeaderCLIPath: String = bufLicenseHeaderCLIFile.absolutePath
 
-tasks.register<Exec>("installBuf") {
+tasks.register("configureBuf") {
     description = "Installs the Buf CLI."
-    environment("GOBIN", bufCLIFile.parentFile.absolutePath)
-    outputs.file(bufCLIFile)
-    commandLine("go", "install", "github.com/bufbuild/buf/cmd/buf@latest")
+    File(buf.asPath).setExecutable(true)
```

Comment:
Instead of needing to download all the buf dependencies and build the CLI, we just download the Maven artifact and make it executable. This also has the benefit of including the artifact in the existing gradle CI cache.